.github/workflows/static-analysis.yml

```diff
@@ -0,0 +1,30 @@
+name: Static code analysis
+
+on: push
+
+jobs:
+
+  flake8:
+    runs-on: ubuntu-latest
+
+    steps:
+      - uses: actions/checkout@v4
+
+      - uses: actions/setup-python@v5
+        with:
+          python-version: 3.12
+
+      - run: |
+          python -m pip install --upgrade pip
+          python -m pip install flake8 flake8-import-order flake8-builtins  # FIXME add flake8-blind-except
+      - run: flake8 --max-line-length=120 --import-order-style=pycharm --statistics
```

Comment:
Our newer repositories (i.e., its_live_monitoring) are [using ruff here instead](https://github.com/ASFHyP3/its-live-monitoring/blob/develop/.github/workflows/static-anaysis.yml). Are we happy with flake8 here, or do we want to switch?